mhs-reference-implementation/mhs/routing/abstract_cache_adapter.py

```diff
@@ -0,0 +1,30 @@
+import abc
+from typing import Optional, Dict
+
+FIFTEEN_MINUTES_IN_SECONDS = 900
+
+
+class AbstractMHSCacheAdaptor(abc.ABC):
+
+    def __init__(self, expiry_time=FIFTEEN_MINUTES_IN_SECONDS):
+        """
+        :param expiry_time: Time for a value to expire in seconds
+        """
+        if expiry_time < 0:
+            raise ValueError('Expiry time must not be non-negative')
+        self.expiry_time = expiry_time
+
+    @abc.abstractmethod
+    async def retrieve_mhs_attributes_value(self, ods_code: str, interaction_id: str) -> Optional[Dict]:
```

Comment:
I've updated the `add_cache_value` in the abstract class to be async.

My original thought was that they should all be `async`, since depending on the implementation of the cache it could involve blocking calls.  Does this seem a valid assumption?

Reply:
I think so.